proxy_map.go

```diff
@@ -0,0 +1,1023 @@
+	return m.tryLock(key, 0, timeout.Milliseconds())
+}
+
+// TryLockWithLeaseTimeout tries to acquire the lock for the specified key.
+// The current thread becomes disabled for thread scheduling purposes and lies
+// dormant until one of the followings happens:
+// - The lock is acquired by the current thread, or
+// - The specified waiting time elapses.
+// Lock will be released after lease time passes.
+func (m *Map) TryLockWithLeaseTimeout(key interface{}, lease time.Duration, timeout time.Duration) (bool, error) {
+	return m.tryLock(key, lease.Milliseconds(), timeout.Milliseconds())
+}
+
+// TryPut tries to put the given key and value into this map and returns immediately.
+func (m *Map) TryPut(key interface{}, value interface{}) (bool, error) {
+	return m.tryPut(key, value, 0)
+}
+
+// TryPut tries to put the given key and value into this map and waits until operation is completed or timeout is reached.
+func (m *Map) TryPutWithTimeout(key interface{}, value interface{}, timeout time.Duration) (bool, error) {
+	return m.tryPut(key, value, timeout.Milliseconds())
+}
+
+// TryRemove tries to remove the given key from this map and returns immediately.
+func (m *Map) TryRemove(key interface{}) (interface{}, error) {
+	return m.tryRemove(key, 0)
+}
+
+// TryRemoveWithTimeout tries to remove the given key from this map and waits until operation is completed or timeout is reached.
+func (m *Map) TryRemoveWithTimeout(key interface{}, timeout time.Duration) (interface{}, error) {
+	return m.tryRemove(key, timeout.Milliseconds())
+}
+
+// Unlock releases the lock for the specified key.
+func (m *Map) Unlock(key interface{}) error {
+	if keyData, err := m.validateAndSerialize(key); err != nil {
+		return err
+	} else {
+		refID := m.refIDGenerator.NextID()
+		request := codec.EncodeMapUnlockRequest(m.name, keyData, m.lockID, refID)
+		_, err = m.invokeOnKey(m.ctx, request, keyData)
+		return err
+	}
+}
+
+// ListenEntryNotification adds a continuous entry listener to this map.
+func (m *Map) ListenEntryNotification(config MapEntryListenerConfig, handler EntryNotifiedHandler) (string, error) {
+	flags := makeListenerFlags(&config)
+	subscriptionID := int(m.refIDGenerator.NextID())
+	if err := m.listenEntryNotified(flags, config.IncludeValue, config.Key, config.Predicate, subscriptionID, handler); err != nil {
+		return "", err
+	}
+	return strconv.Itoa(subscriptionID), nil
+}
+
+// UnlistenEntryNotification removes the specified entry listener.
+func (m *Map) UnlistenEntryNotification(subscriptionID string) error {
+	if subscriptionIDInt, err := strconv.Atoi(subscriptionID); err != nil {
+		return fmt.Errorf("invalid subscription ID: %s", subscriptionID)
+	} else {
+		m.userEventDispatcher.Unsubscribe(EventEntryNotified, subscriptionIDInt)
+		return m.listenerBinder.Remove(m.name, subscriptionIDInt)
+	}
+}
+
+func (m *Map) addIndex(indexConfig types.IndexConfig) error {
+	request := codec.EncodeMapAddIndexRequest(m.name, indexConfig)
```

Comment:
IndexConfig validation is missing. See https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java#L1326

Reply:
Thanks, added that.